lib/multicallUtils.ts

```diff
@@ -0,0 +1,30 @@
+
+const MULTICALL_BATCH_SIZE = 50;
+
+export async function batchedMulticall<T>(
+  client: any,
+  contracts: any[],
+  batchSize = MULTICALL_BATCH_SIZE,
+  delayMs = 500
+): Promise<T[]> {
```

Comment:
i think the correct type would be `Promise<(T | undefined)[]>`? because some requests could fail and `results` will be filled with `undefined`